### ISSUE TITLE
Inconsistency between Algorithm 1 and implementation in ScanNet/nuScenes training scripts

### ISSUE THREAD

**Issue by Herry0w0** [vLAR-group/LogoSP#10]
Hi authors,

First of all, thank you for the fantastic work and for open-sourcing the codebase! I have been studying the `LogoSP` paper and the code, but I noticed a discrepancy regarding the **Progressive Growing** strategy (Algorithm 1) in the training scripts.

In the paper (Section 3.2 and Algorithm 1), it is stated that the framework dynamically updates the pseudo-labels by growing the superpoints (decreasing ) and re-clustering the global patterns every  epochs.

I found that this is perfectly implemented in `train_Seg_S3DIS.py`:

```python
# In train_Seg_S3DIS.py
for epoch in range(1, args.max_epoch + 1):
    if (epoch - 1) % 10 == 0:
        print('Current SP', current_sp)
        classifier = cluster(args, logger, cluster_loader, current_sp, model)
        current_sp -= 10 
        # ...

```

However, when checking the scripts for larger datasets, namely `train_Seg_ScanNet.py` and `train_Seg_nuScenes.py`, this progressive growing logic seems to be missing.
In both scripts, the `cluster()` function is only called **once** before the training loop, with `current_sp` strictly fixed (to 40 for ScanNet and 20 for nuScenes). During the entire `for epoch in range(...)` loop, the pseudo-labels are never updated, and the network seems to only fit the static pseudo-labels generated at epoch 0:

```python
# In train_Seg_ScanNet.py (and similarly in nuScenes)
current_sp = 40
classifier = cluster(args, logger, cluster_loader, current_sp, model) # Called only once

for epoch in range(1, args.max_epoch + 1):
    # No cluster() update or current_sp reduction here
    train(train_loader, logger, model, optimizer, loss, epoch, scheduler, classifier)

```

**My questions are:**

1. Is this omission in `train_Seg_ScanNet.py` and `train_Seg_nuScenes.py` intentional? Was it simplified due to the high computational complexity of Graph Fourier Transform (eigenvalue decomposition) on large-scale/sparse datasets, or to prevent noise accumulation (pseudo-label drift) during training?
2. Were the reported state-of-the-art results for ScanNet and nuScenes in the paper achieved using these static clustering scripts, or is there a missing progressive growing version for these datasets?

Any clarification would be greatly appreciated. Thanks again for your excellent contribution to the unsupervised 3D segmentation community!

**Comment by zihui0930** [comment]
Hi, @Herry0w0 

This work has a distillation stage, which provides better network initialization than GrowSP. We found that in this case, the superpoint growing can be much more aggressive, even complete in one step, in some datasets, e.g., ScanNet and nuScenes. So, we simplify the growing process. 

The current code has very close performance to that in the paper. Before submitting to CVPR, we gradually grow superpoints in training, but later found that a more aggressive approach could also be effective, with both methods yielding results very close to the paper. For simplicity, we use the more aggressive one. 

regards,
zihui

**Comment by Herry0w0** [comment]
Thank you for your reply!
